### PR TITLE
Fix Missing helptext in status bar when mouse is above submenu-entry

### LINKED ISSUE #24842: Missing helptext in status bar when mouse is above submenu-entry
# Description

As the title says, you don't get the help-text in the status bar when the mouse is over a submenu-entry.
All the other menu-entries do show the helptext, so I am not sure if its my error or a problem in the library.

# To Reproduce:

- edit the samples\statbar\statbar.cpp
- add in line 359 a helptext: 'statbarStyleMenu);' --> 'statbarStyleMenu, "help Status bar style");'
- add in line 403 a helptext: 'statbarPaneStyleMenu);' --> 'statbarPaneStyleMenu, "help Field style");'

Btw already in 2009 there was a question about this, but no answers:
"wxMenu::AppendSubMenu help string not displaying"
[viewtopic.php?p=111705#p111705](https://forums.wxwidgets.org/viewtopic.php?p=111705#p111705)

# Platform and version information:
  - wxMSW 
  - wxWidgets v3.2.6 (same in v3.2.4, v3.2.5)
  - Win10

### PR DESCRIPTION
**Pull request by mxdamien**
closes #24842

### CODE REVIEW

File: src/msw/window.cpp [hunk 1]
```diff
@@ -2577,6 +2579,38 @@ wxMenu* wxWindowMSW::MSWFindMenuFromHMENU(WXHMENU hMenu)
     return nullptr;
 }
 
+wxMenuItem* wxWindowMSW::MSWFindMenuItemFromHMENU(WXHMENU hMenu, int item)
+{
+    MENUITEMINFO mii;
+    mii.cbSize = sizeof(MENUITEMINFO);
```
Comment (vadz) [inline src/msw/window.cpp:2585]: Minor, but this is a bit shorter and more consistent with the rest of our code:
```suggestion
    WinStruct<MENUITEMINFO> mii;
```

File: include/wx/frame.h [hunk 2]
```diff
@@ -233,6 +233,9 @@ class WXDLLIMPEXP_CORE wxFrameBase : public wxTopLevelWindow
     // return false if there is no help for such item
     bool ShowMenuHelp(int helpid);
 
+    // show the help string for a menu item directly without finding it by id
+    bool ShowMenuHelp(const wxString& helpString);
```
Comment (vadz) [inline include/wx/frame.h:237]: I'm not really sure we need this function, it seems to be exactly the same as `DoGiveHelp()`, isn't it?

File: src/common/framecmn.cpp [hunk 3]
```diff
@@ -343,7 +343,15 @@ void wxFrameBase::OnMenuHighlight(wxMenuEvent& event)
 {
     event.Skip();
 
-    (void)ShowMenuHelp(event.GetMenuId());
+    wxMenuItem* menuItem = event.GetMenuItem();
+    if( menuItem )
```
Comment (vadz) [inline src/common/framecmn.cpp:347]: Minor, but I'd write this as
```suggestion
    if ( wxMenuItem* menuItem = event.GetMenuItem() )
```

**Review by vadz — COMMENTED** [review]
Thanks David, this looks good to me and I'd just like to simplify it a tiny little bit before merging it.

File: include/wx/frame.h (hunk 2 above)
Comment (mxdamien) [reply include/wx/frame.h:237]: You're right. Of course DoGiveHelp can be used directly here.

**Comment by mxdamien** [comment]
Great! Thank you very much for the review Vadim. I agree on all your remarks

File: include/wx/event.h [hunk 4]
```diff
@@ -48,6 +48,7 @@ class WXDLLIMPEXP_FWD_BASE wxEventFilter;
     class WXDLLIMPEXP_FWD_CORE wxReadOnlyDC;
     class WXDLLIMPEXP_FWD_CORE wxWindow;
     class WXDLLIMPEXP_FWD_CORE wxWindowBase;
+    class WXDLLIMPEXP_FWD_CORE wxMenuItem;
```
Comment (PBfordev) [inline include/wx/event.h:51]: nitpick: The classes are declared in the alphabetical order, so `wxMenuItem` should be moved after `wxMenu`.

Comment (mxdamien) [reply include/wx/event.h]: Thank you !

**Pull request merged**